### DISCUSSION TITLE
Retirement of renewables

### DISCUSSION THREAD

**Discussion by wanyingw193** [NREL/ReEDS-2.0#119] [Q&A: Data]
Hi ReEDS team, when looking at the ReEDS technical report 2021, it wrote, "Renewable energy generator and battery retirements are (by default) based on assumed lifetimes. Once a generator has reached its lifetime, it is retired. "

But I also noticed that in the d_solveprep.gms, 
noretire(i) "technologies that will never be retired" /distpv, can-imports/; 
noretire(i)$[(storage_standalone(i) or hyd_add_pump(i))] = yes; 

Questions:
(1) Does this mean that the battery, hydro-pump and distpv will never retire no matter what lifetime they have?
(2) Is maxage(i) the parameter that determines the retirement of renewable energy?

**Comment by wesleyjcole** [comment]
"noretire" sets which technologies are not allowed to be endogenously retired, so technologies included in noretire cannot be retired before their lifetime.  They will still retire when they meet their lifetime.

Yes, maxage controls the maximum lifetimes of technologies.
